### PR TITLE
OpenGL: Take cached viewport sub-rect into account for scissor

### LINKED ISSUE #1938: Several games with partial/cut-off rendering on the bottom screen
Affected game: Kung Fu Panda - SLL
Appears to be a regression from #1933. Tested with Nvidia GPU.

kp1
kp2
[LOG](http://clipboard.space/clip/ggye8KrWjnHPpg551keq)

### PR DESCRIPTION
**Pull request by yuriks**
Fixes #1938

### CODE REVIEW

**Comment by neobrain** [comment]
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2016-09-16T07:27:18Z: https://github.com/citra-emu/citra/compare/8bf09b5302dc7717477f6c3b5c1f0ea8f2c80717...yuriks:cc40ebda471b7b992247efda1211ac9086c7be53

**Comment by bunnei** [comment]
LGTM

File: src/video_core/renderer_opengl/gl_shader_gen.cpp [hunk 1]
```diff
@@ -597,10 +597,10 @@ vec4 secondary_fragment_color = vec4(0.0);
         if (state.scissor_test_mode == Regs::ScissorMode::Include)
             out += "!";
         // x2,y2 have +1 added to cover the entire pixel area
```
Comment (PeterTh) [inline src/video_core/renderer_opengl/gl_shader_gen.cpp]: This comment should probably be moved to gl_rasterizer where the scissor computation actually happens now.

Comment (yuriks) [inline src/video_core/renderer_opengl/gl_shader_gen.cpp]: Thanks. Updated.

**Pull request merged**